core/metacling/src/rootclingTCling.cxx

```diff
@@ -20,6 +20,11 @@
 
 #undef R__DLLEXPORT
 
+#ifdef __FreeBSD__
+char*  __progname;
+char** environ;
```

Comment:
These are `extern`, no?

Reply:
possibly, this is a "legacy" fix from the port maintainer, it is needed and sufficient (it seems). I think maybe for some reason the symbols needed to be visible for lld, possibly some subtle difference with ld.